src/ui/components/results/paginationcomponent.js

```diff
@@ -164,7 +164,13 @@ export default class PaginationComponent extends Component {
   }
 
   updatePage (offset) {
-    this._onPaginate();
+    const results = this.core.globalStorage.getState(StorageKeys.VERTICAL_RESULTS) || {};
+    const oldOffset = this.core.globalStorage.getState(StorageKeys.SEARCH_OFFSET) || 0;
+    const limit = this.core.globalStorage.getState(StorageKeys.SEARCH_CONFIG).limit;
+    const oldPageNumber = (oldOffset / limit) + 1;
+    const pageNumber = (offset / limit) + 1;
+    const maxPageCount = Math.trunc((results.resultsCount - 1) / limit) + 1;
```

Comment:
We recompute the limit many times in this class. But, this quantity is effectively static (once the page has loaded). To avoid inefficiency, maybe we should compute limit once, in the constructor. I think we should also make a helper to compute maxPageCount. That way, we can refer to the helper and not have this logic written out explicitly in multiple places.